drivers/mipi_dbi/mipi_dbi_stm32_fmc.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_fmc_mipi_dbi
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mipi_dbi.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mipi_dbi_stm32_fmc, CONFIG_MIPI_DBI_LOG_LEVEL);
+
+#define FMC_DATA_ADDR_OFFSET 0x02
+
+struct mipi_dbi_stm32_fmc_config {
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	/* Power GPIO */
+	const struct gpio_dt_spec power;
+	uint16_t *register_addr;
+	uint16_t *data_addr;
+};
+
+struct mipi_dbi_stm32_fmc_data {
+};
+
+int mipi_dbi_stm32_fmc_command_write(const struct device *dev,
```

Comment:
I tried to do everything you said. I only did try to check if the configurations match between the fmc and the display controler: I'm really not sure about how to reconfigure the fmc timings (there is multiple settings possible that will impact the overall frequency) so I just check that the frequency of the writes from the fmc is lower than the max frequency of the display controler (from my testing we can go higher in frequency, but I suppose there is just a bit of margin between the documentation and the reality ?).

For the access mode, as I only have a board with the screen using intel 8080 16 bits, the driver will only support this for now, I think it can be configured for the other mode but I can't test it. I only checked that the mode asked is 16 bit, and that the fmc is configured for 16 bits : I could check all configuration values for the fmc, should I do it ?

Reply:
I am ok to limit the driver to 16 bits if that is all we can test

Reply:
I mean I could check all configurations values of the fmc (set in the devicetree)
```
			st,control = <STM32_FMC_DATA_ADDRESS_MUX_DISABLE
				STM32_FMC_MEMORY_TYPE_SRAM
				STM32_FMC_NORSRAM_MEM_BUS_WIDTH_16
				STM32_FMC_BURST_ACCESS_MODE_DISABLE
				STM32_FMC_WAIT_SIGNAL_POLARITY_LOW
				STM32_FMC_WAIT_TIMING_BEFORE_WS
				STM32_FMC_WRITE_OPERATION_ENABLE
				STM32_FMC_WAIT_SIGNAL_DISABLE
				STM32_FMC_EXTENDED_MODE_DISABLE
				STM32_FMC_ASYNCHRONOUS_WAIT_DISABLE
				STM32_FMC_WRITE_BURST_DISABLE
				STM32_FMC_CONTINUOUS_CLOCK_SYNC_ONLY
				STM32_FMC_WRITE_FIFO_DISABLE
				STM32_FMC_PAGE_SIZE_NONE>;
			st,timing = <1 1 32 0 2 2 STM32_FMC_ACCESS_MODE_A>;
```

Now the code check the timings, and the bus size. I'm not totally sure which other configuration values are important, but I could look into it if it necessary ?

Reply:
I think this is ok. I'm not familiar with the FMC, but it seems that timings and bus size would be the most important- unless it is possible to configure the FMC for Intel 8080 and Motorola 6800 modes. If only one mode is possible between those, then checking bus size and timings should be enough.